toolkit/scss/forms/_questions.scss

```diff
@@ -12,6 +12,11 @@
   .hint {
     margin-bottom: 0;
   }
+
+  // This preserves the vertical margins set on the .question-headings
+  legend {
```

Comment:
For the record, this was resolved by @pcraig3 working with @ralph-hawkins to make the vertical spacing for all questions consistent, whatever elements are present about the field.